arangod/Aql/IResearchViewExecutor.cpp

```diff
@@ -680,14 +672,48 @@ bool IResearchViewExecutorBase<Impl, Traits>::writeRow(ReadContext& ctx,
                                                        LocalDocumentId const& documentId,
                                                        LogicalCollection const& collection) {
   TRI_ASSERT(documentId.isSet());
-  bool writeDocOk;
-  if constexpr (Traits::MaterializeType == MaterializeType::Materialized) {
+  auto writeDocOk = false;
+  auto tryWriteDoc = false;
+  if constexpr (Traits::MaterializeType == MaterializeType::Materialize) {
+    tryWriteDoc = true;
     // read document from underlying storage engine, if we got an id
     writeDocOk = collection.readDocumentWithCallback(infos().getQuery().trx(), documentId, ctx.callback);
-  } else {
+  } else if ((Traits::MaterializeType & MaterializeType::LateMaterialize) == MaterializeType::LateMaterialize) {
+    tryWriteDoc = true;
     // no need to look into collection. Somebody down the stream will do materialization. Just emit LocalDocumentIds
     writeDocOk = writeLocalDocumentId(ctx, documentId, collection);
   }
+  if (!tryWriteDoc || writeDocOk) {
+    if constexpr ((Traits::MaterializeType & MaterializeType::UseStoredValues) == MaterializeType::UseStoredValues) {
+      auto const& columnsFieldsRegs = infos().getOutNonMaterializedViewRegs();
+      TRI_ASSERT(!columnsFieldsRegs.empty());
+      auto max = (--columnsFieldsRegs.cend())->first;
+      TRI_ASSERT(max >= IResearchViewNode::SortColumnNumber);
+      auto columnFieldsRegs = columnsFieldsRegs.cbegin();
+      if (IResearchViewNode::SortColumnNumber == columnFieldsRegs->first) {
+        ++max;
```

Comment:
why don't to move `++max` under second `if (IResearchViewNode::SortColumnNumber == columnFieldsRegs->first) {`?